Add Image#top_level_in_collection? with specs, and pass to common index

bmquinn opened the pull request.

kdid commented:
@bmquinn - the indexing part of this looks fine, but it needs to be triggered when a work has children added to it. (i.e. a nested index of child works kicks off) - otherwise that `top_level` t/f is not updated on the children when they are added/removed to/from the parent.

bmquinn commented:
@kdid I added an actor override to call `update_common_index` on child works at the appropriate time. Also updated `Image#top_level_in_collection?` to be smarter than `in_works.empty?`. MBK and I discussed the location of the `top_level_in_collection` in the indexed document and we think it should be okay on the Glaze side.

kdid commented:
@bmquinn - Adding a child to a work works well for me locally, but when I remove a child I'm getting an error. 

<img width="1087" alt="Screen Shot 2019-08-15 at 11 27 12 AM" src="https://user-images.githubusercontent.com/6372022/63110374-e7472880-bf50-11e9-8e99-339b5256d875.png">
